File: shared/src/commonMain/kotlin/com/example/tp4/Hand.kt
package com.example.tp4

class Hand(val cards: MutableList<Card>) {

    override fun toString(): String {
        return cards.joinToString("") { it.toString() }
    }

    operator fun get(index: Int): Card {
        return cards[index]
    }

    operator fun set(index: Int, card: Card) {
        cards[index] = card
    }
}

File: web/src/game/Hand.ts
import { Card } from "./Card";

export class Hand {
  cards: Card[];

  constructor(cards: Card[]) {
    this.cards = cards;
  }

  toString(): string {
    return this.cards.map((c) => c.toString()).join("");
  }

  get(index: number): Card {
    return this.cards[index];
  }

  set(index: number, card: Card): void {
    this.cards[index] = card;
  }
}
